### ISSUE TITLE
When loading a form with `instance` set to an object, existing data isn't displayed

### ISSUE THREAD

**Issue by vtbassmatt** [OmenApps/django-tomselect#4]
When I try to edit an existing model instance, the Tom-Select widget isn't prepopulated with the current data the way the default Django `Select` widget does.

Default Django `Select` widget:
<img width="423" alt="default Django Select widget showing prepopulated data" src="https://github.com/jacklinke/django-tomselect/assets/714283/be3c6c0e-9a39-4433-b381-cc19f18c3fc2">

Swapping out for this Tom-Select widget:
<img width="352" alt="Tom-Select widget with no prepopulated data" src="https://github.com/jacklinke/django-tomselect/assets/714283/037249e9-f82c-49ac-9f12-0e0af816de10">

(I don't show it here in the issue, but clicking the dropdown does correctly load all the `City` objects.)

<details>
<summary>Repro files</summary>

```python
# app/models.py
from django.db import models

class City(models.Model):
    name = models.CharField(max_length=200)
    def __str__(self):
        return self.name

class Place(models.Model):
    name = models.CharField(max_length=200)
    city = models.ForeignKey(City, on_delete=models.PROTECT)
    def __str__(self):
        return self.name
```

```python
# app/forms.py
from app.models import Place, City
from django import forms
from django_tomselect.widgets import TomSelectWidget


class PlaceForm(forms.ModelForm):
    class Meta:
        model = Place
        fields = ('name', 'city')
        widgets = {'city': TomSelectWidget(City, 'city-auto'),}
```

```python
# app/views.py
from django.shortcuts import render
from .forms import PlaceForm
from .models import Place

def index(request):
    instance = Place.objects.get(pk=1)
    form = PlaceForm(instance=instance)
    return render(request, 'index.html', {'form': form})
```

```python
# project/urls.py
from django.contrib import admin
from django.urls import path
from django_tomselect.views import AutocompleteView
from app import views

urlpatterns = [
    path('admin/', admin.site.urls),
    path('tom/', AutocompleteView.as_view(), name='city-auto'),
    path('', views.index),
]
```

```html
<!-- app/templates/index.html -->
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Django Tom Select Demo</title>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-kenU1KFdBIe4zVF0s0G1M5b4hcpxyD9F7jL+jjXkk+Q2h455rYXK/7HAuoJl+0I4" crossorigin="anonymous"></script>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-rbsA2VBKQhggwzxH7pPCaAqO46MgnOM80zW1RWuH61DGLwZJEdK2Kadq2F9CUG65" crossorigin="anonymous">
    {{ form.media }}
</head>
<body>
<div class="container">
    <form>
        {% csrf_token %}
        {{ form.as_div }}
        <button type="submit" class="btn btn-success">Save</button>
    </form>
</div>
</body>
</html>
```

```python
# interactive session to populate data
from app.models import Place, City
raleigh = City(name='Raleigh')
raleigh.save()
Place(name='Trophy', city=raleigh).save()
```
</details>

**Comment by vtbassmatt** [comment]
Ah, after some more poking, it's because of https://github.com/jacklinke/django-tomselect/blob/c777a84cc31c5ec3af4909a5cec758fa26d521c9/src/django_tomselect/widgets.py#L78-L79 (`optgroups` always returns `[]`). I understand why that line wants to not read the entire queryset, but I'll see if I can patch it to return the currently-selected data.

**Comment by vtbassmatt** [comment]
`django-autcomplete-light` (another MIT-licensed improvement on standard Select widgets) has this implementation: https://github.com/yourlabs/django-autocomplete-light/blob/2e51796eefe37a8efa6b66420e3e355cb73edc91/src/dal/widgets.py#L129-L145

It seems to work here as well. I'll put up a PR.

**Comment by jacklinke** [comment]
Resolved in 2024.12.2
